src/shell/Job.ts

```diff
@@ -113,7 +113,19 @@ export class Job extends EmitterWithUniqueID implements TerminalLikeDevice {
         if (typeof input === "string") {
             text = input;
         } else {
-            text = input.ctrlKey ? String.fromCharCode(input.keyCode - 64) : normalizeKey(input.key, this.screenBuffer.cursorKeysMode);
+            if (input.ctrlKey) {
+                text = String.fromCharCode(input.keyCode - 64);
+            } else if (input.altKey) {
+                let char = String.fromCharCode(input.keyCode);
```

Comment:
From my testing `input.key` is the special char macOS generates, ∂ for alt+d for example.